### PR TITLE
[Sharded DB][2] Add 2 new schemas for statevalue using key hash

### PR DESCRIPTION
**Pull request by areshand**
## Description

Plan to store the state kv in two new schemas when sharding config is enabled.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by grao1991 — APPROVED** [review]

**Comment by areshand** [comment]
@msmouse take a look?  mostly the new names. I am trying to create small diffs

File: storage/aptosdb/src/schema/mod.rs [hunk 1]
```diff
@@ -46,7 +48,10 @@ pub const LEDGER_INFO_CF_NAME: ColumnFamilyName = "ledger_info";
 pub const STALE_NODE_INDEX_CF_NAME: ColumnFamilyName = "stale_node_index";
 pub const STALE_NODE_INDEX_CROSS_EPOCH_CF_NAME: ColumnFamilyName = "stale_node_index_cross_epoch";
 pub const STALE_STATE_VALUE_INDEX_CF_NAME: ColumnFamilyName = "stale_state_value_index";
+pub const STALE_STATE_VALUE_INDEX_WITH_KEY_HASH_CF_NAME: ColumnFamilyName =
+    "stale_state_value_index_with_key_hash";
 pub const STATE_VALUE_CF_NAME: ColumnFamilyName = "state_value";
+pub const STATE_VALUE_WITH_KEY_HASH_CF_NAME: ColumnFamilyName = "state_value_with_key_hash";
```
Comment (msmouse) [inline storage/aptosdb/src/schema/mod.rs:54]: state_value_by_key_hash

File: storage/aptosdb/src/schema/mod.rs [hunk 2]
```diff
@@ -46,7 +48,10 @@ pub const LEDGER_INFO_CF_NAME: ColumnFamilyName = "ledger_info";
 pub const STALE_NODE_INDEX_CF_NAME: ColumnFamilyName = "stale_node_index";
 pub const STALE_NODE_INDEX_CROSS_EPOCH_CF_NAME: ColumnFamilyName = "stale_node_index_cross_epoch";
 pub const STALE_STATE_VALUE_INDEX_CF_NAME: ColumnFamilyName = "stale_state_value_index";
+pub const STALE_STATE_VALUE_INDEX_WITH_KEY_HASH_CF_NAME: ColumnFamilyName =
```
Comment (msmouse) [inline storage/aptosdb/src/schema/mod.rs:51]: stale_state_value_by_key_hash_index

**Review by msmouse — COMMENTED** [review]
please update commit comment

I think _by_key_hash is the right naming.. could you rename?

File: storage/aptosdb/src/schema/stale_state_value_index_by_key_hash/mod.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+//!
+//! An index entry in this data set has 3 pieces of information:
+//!     1. The version since which a state value (in another data set) becomes stale, meaning,
+//! replaced by an updated value.
+//!     2. The version this state value was updated identified by the state key.
+//!     3. The state_key to identify the stale state value.
+//!
+//! ```text
+//! |<-------------------key------------------->|
+//! | stale_since_version | version | state_key_hash |
```
Comment (msmouse) [inline storage/aptosdb/src/schema/stale_state_value_index_by_key_hash/mod.rs:15]: nit: would be nice to see them aligned, lol

File: storage/aptosdb/src/schema/state_value_by_key_hash/mod.rs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        Ok(encoded)
+    }
+
+    fn decode_key(data: &[u8]) -> Result<Self> {
+        const VERSION_SIZE: usize = size_of::<Version>();
+
+        ensure_slice_len_gt(data, VERSION_SIZE)?;
+        let hash_len = data.len() - VERSION_SIZE;
+        let state_key_hash: HashValue = HashValue::from_slice(&data[..hash_len])?;
+        let version = !(&data[hash_len..]).read_u64::<BigEndian>()?;
```
Comment (msmouse) [inline storage/aptosdb/src/schema/state_value_by_key_hash/mod.rs:51]: let's ensure the exact len now.. (32 + 8), and decode the hash from `&data[..HashValue::LENGTH]`

**Review by msmouse — APPROVED** [review]

**Pull request merged**